Content.Shared/Xenoarchaeology/Artifact/XAT/Components/XATCompNearbyComponent.cs

```diff
@@ -0,0 +1,24 @@
+using Robust.Shared.GameStates;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom;
+
+namespace Content.Shared.Xenoarchaeology.Artifact.XAT.Components;
+
+/// <summary>
+/// This is used a XAT that activates when an entity fulfilling the given whitelist is nearby the artifact.
+/// </summary>
+[RegisterComponent, NetworkedComponent, Access(typeof(XATCompNearbyComponent)), AutoGenerateComponentState]
+public sealed partial class XATCompNearbyComponent : Component
+{
+    /// <summary>
+    /// Component name that is required to activate trigger.
+    /// Is spelled without 'Component' suffix.
+    /// </summary>
+    [DataField(customTypeSerializer: typeof(ComponentNameSerializer)), AutoNetworkedField]
+    public string RequireComponentWithName = "Item";
```

Comment:
Docstring says there's a list but there's only a required component here

Reply:
Do we need it to have like list of thouse?... It should still be validated then, i'm not sure how i can do validation for list... It looks like an important cog